### ISSUE TITLE
Random crash while running `ekam -c`

### ISSUE THREAD

**Issue by jparyani** [sandstorm-io/ekam#1]
The following happened randomly while I was doing a bunch of `git rebase` commands in the sandstorm directory while running `make continuous`

```
install: sandstorm/sandstorm.ekam-manifest
terminate called after throwing an instance of 'ekam::OsError'
  what():  inotify_rm_watch(inotify): Invalid argument
Aborted
Makefile:139: recipe for target 'continuous' failed
make: *** [continuous] Error 134
```

**Comment by kentonv** [comment]
Yeah, inotify is notoriously difficult, so ekam sometimes barfs when a lot of stuff changes at once.

At some point I want to rewrite things to be in terms of KJ primitives, and add an inotify primitive to KJ with a sane interface, after which these issues should be gone. In the meantime it doesn't happen too often in my experience...
